shared/actions/chat/inbox.js

```diff
@@ -724,35 +728,26 @@ function* _inboxSynced(action: ChatGen.InboxSyncedPayload): Saga.SagaGenerator<a
   const conversation = Constants.getSelectedConversationStates(state)
   if (conversation) {
     const inbox = Constants.getInbox(state, selectedConversation)
+    const lastMessageIDWeAreShowing = Constants.lastMessageID(state, selectedConversation)
+
+    // Check the data we are given first, then our state
+    const knownMaxMessageID: ?number = latestMaxMsgID[selectedConversation] || (inbox ? inbox.maxMsgID : null)
 
-    const messageKeys = Constants.getConversationMessages(state, selectedConversation)
-    const lastMessageKey = messageKeys.last()
     let numberOverride
-    if (lastMessageKey) {
-      const lastMessage = Constants.getMessageFromMessageKey(state, lastMessageKey)
+    if (lastMessageIDWeAreShowing && knownMaxMessageID) {
+      const lastRpcMessageIdWeAreShowing = Constants.messageIDToRpcMessageID(lastMessageIDWeAreShowing)
+
       // Check to see if we could possibly be asking for too many messages
-      if (
-        lastMessage &&
-        typeof lastMessage.rawMessageID === 'number' &&
-        lastMessage.rawMessageID &&
-        inbox &&
-        inbox.maxMsgID
-      ) {
-        numberOverride = inbox.maxMsgID - lastMessage.rawMessageID
-
-        if (numberOverride > tooManyMessagesToJustAppendOnStale) {
-          console.log(
-            'Doing a full load due to too many old messages',
-            inbox.maxMsgID - lastMessage.rawMessageID
-          )
-          yield Saga.all([
-            Saga.put(ChatGen.createClearMessages({conversationIDKey: selectedConversation})),
-            yield Saga.put(
-              ChatGen.createLoadMoreMessages({conversationIDKey: selectedConversation, onlyIfUnloaded: false})
-            ),
-          ])
-          return
-        }
+      numberOverride = knownMaxMessageID - lastRpcMessageIdWeAreShowing
```

Comment:
This could still have worked but I think it would have been a more specific case.

The bug meant that numberOverride was 0 because it was checking `inbox.maxMsgID` which was the same as `lastMessageIDWeAreShowing`. That resulted in loading only the default of 50 messages on the LoadMoreMessages call and leading to a conv being out of sync.